### ISSUE TITLE
linter: false positive for `unicorn/prefer-array-some`

### ISSUE THREAD

**Issue by pumano** [oxc-project/oxc#7404]
### What version of Oxlint are you using?

0.12.0

### What command did you run?

oxlint -c oxlintrc.json --quiet

### What does your `.oxlint.json` config file look like?

```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["unicorn"],
  "rules": {
    "unicorn/prefer-array-some": "error"
  }
}
```

### What happened?

That rule not properly handle `findIndex` and `findLastIndex` and negative one or zero comparison

Few months ago [commit](https://github.com/sindresorhus/eslint-plugin-unicorn/commit/10568ab392207010ca0da4bbb8e4720e3c23d704#diff-d8c5c430391481edee3ce97981410f286031482d3666489f51bb855445fe0295R243) landed and add proper support.

I prepare PR for that support, but without fixer. I add `TODO` marks for it. Hope someone can add it later.
